lib/llvmopencl/ImplicitConditionalBarriers.cc

```diff
@@ -171,7 +171,11 @@ ImplicitConditionalBarriers::run(llvm::Function &F,
     // TODO: investigate. It might related to the alloca-converted
     // PHIs. It has a loop that is autoconverted to a b-loop and the
     // conditional barrier is inserted after the loop short cut check.
+#if LLVM_MAJOR < 20
     Barrier::create(Pos->getFirstNonPHI());
+#else
+    Barrier::create(Pos->getFirstInsertionPt());
```

Comment:
Is this correct? It might insert before PHI?

Reply:
It's fine: "Returns an iterator to the first instruction in this block that is suitable for inserting a non-PHI instruction."

Reply:
.... "In particular, it skips all PHIs and LandingPad instructions"

It seems (to me) more correct (in theory) to use `getFirstInsertionPt()`, but since OpenCL does not support exceptions, there shouldn't be any LandingPad instructions, so it should be identical to `getFirstNonPhi()`.